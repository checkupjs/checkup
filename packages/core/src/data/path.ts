export function trimCwd(path: string, cwd: string) {
  return path.replace(`${cwd}/`, '');
}

export function trimAllCwd(paths: string[], cwd: string) {
  return paths.map((path) => trimCwd(path, cwd));
}
