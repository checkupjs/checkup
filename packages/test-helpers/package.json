{
  "name": "@checkup/test-helpers",
  "description": "Checkup's test helper library",
  "version": "0.0.10",
  "author": "Steve Calvert <steve.calvert@gmail.com>",
  "bugs": "https://github.com/checkupjs/checkup/issues",
  "dependencies": {
    "@checkup/core": "0.0.10",
    "fixturify-project": "^2.1.0",
    "handlebars": "^4.7.6",
    "stdout-stderr": "^0.1.13",
    "tmp": "^0.2.0"
  },
  "engines": {
    "node": ">= 12.*"
  },
  "files": [
    "/lib",
    "/static"
  ],
  "homepage": "https://github.com/checkupjs/checkup",
  "license": "MIT",
  "main": "lib/index.js",
  "publishConfig": {
    "access": "public"
  },
  "repository": "https://github.com/checkupjs/checkup",
  "scripts": {
    "build": "tsc --build",
    "test": "jest --passWithNoTests --no-cache"
  },
  "types": "lib/index.d.ts"
}
