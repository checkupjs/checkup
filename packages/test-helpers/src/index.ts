export { stdout } from './stdout';
export { createTmpDir } from './tmp-dir';
export { testRoot } from './test-root';
export { getTaskContext } from './get-task-context';
export { filterPieChartDataForTest } from './validate-chart-results';
export { default as CheckupProject } from './checkup-fixturify-project';
export { default as EmberProject } from './ember-cli-fixturify-project';
export { default as Plugin } from './plugin';
