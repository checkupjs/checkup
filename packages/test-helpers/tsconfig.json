{
  "extends": "../../tsconfig-base.json",
  "compilerOptions": {
    "outDir": "lib",
    "rootDir": "src",
    "types": ["chai", "jest", "node"],
    "esModuleInterop": true
  },
  "references": [
    {
      "path": "../core"
    }
  ],
  "include": ["src/**/*"]
}
