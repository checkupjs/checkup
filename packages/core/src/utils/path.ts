import { isAbsolute, join, relative } from 'path';

export function toAbsolute(baseDir: string, filePath: string) {
  let absolutePath = filePath;

  if (!isAbsolute(filePath)) {
    absolutePath = join(baseDir, filePath);
  }

  return absolutePath;
}

export function toRelative(baseDir: string, filePath: string) {
  let relativePath = filePath;

  if (isAbsolute(relativePath)) {
    relativePath = relative(baseDir, filePath);
  }

  return relativePath;
}
