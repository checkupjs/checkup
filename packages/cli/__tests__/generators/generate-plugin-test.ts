import * as helpers from 'yeoman-test';

import { createTmpDir, testRoot } from '@checkup/test-helpers';

import PluginGenerator from '../../src/generators/plugin';
import { join } from 'path';

describe('plugin generator', () => {
  let tmp: string;

  beforeEach(() => {
    tmp = createTmpDir();
  });

  it('generates plugin with defaults', async () => {
    let dir = await helpers
      .run(PluginGenerator, { namespace: 'checkup:plugin' })
      .cd(tmp)
      .withOptions({
        name: 'my-plugin',
        defaults: true,
      });

    let root = testRoot(dir, 'checkup-plugin-my-plugin');

    expect(root.file('package.json').contents).toMatchSnapshot();
    expect(root.file('README.md').contents).toMatchSnapshot();
    expect(root.file('jest.config.js').contents).toMatchSnapshot();
    expect(root.file('tsconfig.json').contents).toMatchSnapshot();
    expect(root.file('src/index.ts').contents).toMatchSnapshot();
    expect(root.file('src/hooks/register-tasks.ts').contents).toMatchSnapshot();
    expect(root.file('src/types/index.ts').contents).toMatchSnapshot();
    expect(root.directory('__tests__').contents).toMatchSnapshot();
    expect(root.directory('src/results').contents).toMatchSnapshot();
    expect(root.directory('src/tasks').contents).toMatchSnapshot();
  });

  it('generates plugin with unmodified name with defaults', async () => {
    let dir = await helpers
      .run(PluginGenerator, { namespace: 'checkup:plugin' })
      .cd(tmp)
      .withOptions({
        name: 'checkup-plugin-foo',
        defaults: true,
      });

    let root = testRoot(dir, 'checkup-plugin-foo');

    expect(root.file('package.json').contents).toMatchSnapshot();
    expect(root.file('README.md').contents).toMatchSnapshot();
    expect(root.file('jest.config.js').contents).toMatchSnapshot();
    expect(root.file('tsconfig.json').contents).toMatchSnapshot();
    expect(root.file('src/index.ts').contents).toMatchSnapshot();
    expect(root.file('src/hooks/register-tasks.ts').contents).toMatchSnapshot();
    expect(root.file('src/types/index.ts').contents).toMatchSnapshot();
    expect(root.directory('__tests__').contents).toMatchSnapshot();
    expect(root.directory('src/results').contents).toMatchSnapshot();
    expect(root.directory('src/tasks').contents).toMatchSnapshot();
  });

  it('generates plugin into existing directory with defaults', async () => {
    let existingDir = join(tmp, 'checkup-plugin-bar');

    await helpers.run(PluginGenerator, { namespace: 'checkup:plugin' }).cd(tmp).withOptions({
      name: 'checkup-plugin-bar',
      defaults: true,
    });

    let root = testRoot(existingDir);

    expect(root.file('package.json').contents).toMatchSnapshot();
    expect(root.file('README.md').contents).toMatchSnapshot();
    expect(root.file('jest.config.js').contents).toMatchSnapshot();
    expect(root.file('tsconfig.json').contents).toMatchSnapshot();
    expect(root.file('src/index.ts').contents).toMatchSnapshot();
    expect(root.file('src/hooks/register-tasks.ts').contents).toMatchSnapshot();
    expect(root.file('src/types/index.ts').contents).toMatchSnapshot();
    expect(root.directory('__tests__').contents).toMatchSnapshot();
    expect(root.directory('src/results').contents).toMatchSnapshot();
    expect(root.directory('src/tasks').contents).toMatchSnapshot();
  });

  it('generates plugin with custom options', async () => {
    let dir = await helpers
      .run(PluginGenerator, { namespace: 'checkup:plugin' })
      .cd(tmp)
      .withOptions({
        name: 'my-plugin',
      })
      .withPrompts({
        description: 'My custom plugin',
        author: 'scalvert <steve.calvert@gmail.com>',
        repository: 'http://github.com/scalvert/checkup-plugin-my-plugin',
      });

    let root = testRoot(dir, 'checkup-plugin-my-plugin');

    expect(root.file('package.json').contents).toMatchSnapshot();
    expect(root.file('README.md').contents).toMatchSnapshot();
    expect(root.file('jest.config.js').contents).toMatchSnapshot();
    expect(root.file('tsconfig.json').contents).toMatchSnapshot();
    expect(root.file('src/index.ts').contents).toMatchSnapshot();
    expect(root.file('src/hooks/register-tasks.ts').contents).toMatchSnapshot();
    expect(root.file('src/types/index.ts').contents).toMatchSnapshot();
    expect(root.directory('__tests__').contents).toMatchSnapshot();
    expect(root.directory('src/results').contents).toMatchSnapshot();
    expect(root.directory('src/tasks').contents).toMatchSnapshot();
  });
});
