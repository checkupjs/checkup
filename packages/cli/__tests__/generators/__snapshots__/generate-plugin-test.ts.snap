// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`plugin generator generates plugin into existing directory with defaults 1`] = `
"{
  \\"name\\": \\"checkup-plugin-bar\\",
  \\"description\\": \\"Checkup plugin\\",
  \\"version\\": \\"0.0.0\\",
  \\"author\\": \\"\\",
  \\"dependencies\\": {
    \\"@checkup/core\\": \\"^0.0.1\\",
    \\"@oclif/command\\": \\"^1\\",
    \\"@oclif/config\\": \\"^1\\",
    \\"tslib\\": \\"^1\\"
  },
  \\"devDependencies\\": {
    \\"@checkup/test-helpers\\": \\"^0.0.1\\",
    \\"@oclif/dev-cli\\": \\"^1\\",
    \\"@types/jest\\": \\"^25.1.3\\",
    \\"@types/node\\": \\"^13\\",
    \\"jest\\": \\"^25.1.0\\",
    \\"ts-jest\\": \\"^25.2.1\\",
    \\"ts-node\\": \\"^8\\",
    \\"typescript\\": \\"^3.8\\"
  },
  \\"engines\\": {
    \\"node\\": \\">= 12.*\\"
  },
  \\"files\\": [
    \\"/lib\\",
    \\"/oclif.manifest.json\\"
  ],
  \\"keywords\\": [
    \\"checkup-plugin\\",
    \\"oclif-plugin\\"
  ],
  \\"license\\": \\"MIT\\",
  \\"oclif\\": {
    \\"devPlugins\\": [
      \\"@oclif/plugin-help\\"
    ],
    \\"hooks\\": {
      \\"register-tasks\\": \\"./lib/hooks/register-tasks\\"
    }
  },
  \\"repository\\": \\"\\",
  \\"scripts\\": {
    \\"build\\": \\"yarn clean && tsc\\",
    \\"build:watch\\": \\"yarn build -w\\",
    \\"clean\\": \\"rm -rf lib\\",
    \\"postpack\\": \\"rm -f oclif.manifest.json\\",
    \\"prepack\\": \\"rm -rf lib && tsc -b && oclif-dev manifest && oclif-dev readme\\",
    \\"test\\": \\"jest --no-cache\\",
    \\"version\\": \\"oclif-dev readme && git add README.md\\"
  },
  \\"types\\": \\"lib/index.d.ts\\",
  \\"main\\": \\"lib/index.js\\"
}
"
`;

exports[`plugin generator generates plugin into existing directory with defaults 2`] = `
"# checkup-plugin-bar

Checkup plugin

[![oclif](https://img.shields.io/badge/cli-oclif-brightgreen.svg)](https://oclif.io)
[![Version](https://img.shields.io/npm/v/plugin-default.svg)](https://npmjs.org/package/plugin-default)
[![Downloads/week](https://img.shields.io/npm/dw/plugin-default.svg)](https://npmjs.org/package/plugin-default)
[![License](https://img.shields.io/npm/l/plugin-default.svg)](https://github.com/checkupjs/checkup/blob/master/package.json)

<!-- toc -->

# Usage

<!-- usage -->

# Commands

<!-- commands -->
"
`;

exports[`plugin generator generates plugin into existing directory with defaults 3`] = `
"module.exports = {
  testEnvironment: 'node',
  moduleFileExtensions: ['ts', 'js', 'json'],
  transform: { '\\\\\\\\.ts$': 'ts-jest' },
  coverageReporters: ['lcov', 'text-summary'],
  collectCoverageFrom: ['src/**/*.ts'],
  coveragePathIgnorePatterns: ['/templates/'],
  coverageThreshold: {
    global: {
      branches: 100,
      functions: 100,
      lines: 100,
      statements: 100,
    },
  },
  testPathIgnorePatterns: ['__fixtures__'],
};
"
`;

exports[`plugin generator generates plugin into existing directory with defaults 4`] = `
"{
  \\"compilerOptions\\": {
    \\"declaration\\": true,
    \\"importHelpers\\": true,
    \\"module\\": \\"commonjs\\",
    \\"strict\\": true,
    \\"target\\": \\"es2017\\",
    \\"allowSyntheticDefaultImports\\": true,
    \\"sourceMap\\": true,
    \\"outDir\\": \\"lib\\",
    \\"rootDir\\": \\"src\\",
    \\"types\\": [\\"jest\\", \\"node\\"]
  },
  \\"include\\": [\\"src/**/*\\"]
}
"
`;

exports[`plugin generator generates plugin into existing directory with defaults 5`] = `
"export default {};

export * from './types';
"
`;

exports[`plugin generator generates plugin into existing directory with defaults 6`] = `
"import { Hook } from '@oclif/config';

const hook: Hook<'register-tasks'> = async function({ context, tasks }: any) {
  
};

export default hook;
"
`;

exports[`plugin generator generates plugin into existing directory with defaults 7`] = `""`;

exports[`plugin generator generates plugin into existing directory with defaults 8`] = `
Array [
  ".gitkeep",
]
`;

exports[`plugin generator generates plugin into existing directory with defaults 9`] = `
Array [
  ".gitkeep",
]
`;

exports[`plugin generator generates plugin into existing directory with defaults 10`] = `
Array [
  ".gitkeep",
]
`;

exports[`plugin generator generates plugin with JavaScript defaults 1`] = `
"{
  \\"name\\": \\"checkup-plugin-my-plugin\\",
  \\"description\\": \\"Checkup plugin\\",
  \\"version\\": \\"0.0.0\\",
  \\"author\\": \\"\\",
  \\"dependencies\\": {
    \\"@checkup/core\\": \\"^0.0.1\\",
    \\"@oclif/command\\": \\"^1\\",
    \\"@oclif/config\\": \\"^1\\"
  },
  \\"devDependencies\\": {
    \\"@checkup/test-helpers\\": \\"^0.0.1\\",
    \\"@oclif/dev-cli\\": \\"^1\\",
    \\"jest\\": \\"^25.1.0\\"
  },
  \\"engines\\": {
    \\"node\\": \\">= 12.*\\"
  },
  \\"files\\": [
    \\"/lib\\",
    \\"/oclif.manifest.json\\"
  ],
  \\"keywords\\": [
    \\"checkup-plugin\\",
    \\"oclif-plugin\\"
  ],
  \\"license\\": \\"MIT\\",
  \\"oclif\\": {
    \\"devPlugins\\": [
      \\"@oclif/plugin-help\\"
    ],
    \\"hooks\\": {
      \\"register-tasks\\": \\"./lib/hooks/register-tasks\\"
    }
  },
  \\"repository\\": \\"\\",
  \\"scripts\\": {
    \\"postpack\\": \\"rm -f oclif.manifest.json\\",
    \\"prepack\\": \\"oclif-dev manifest && oclif-dev readme\\",
    \\"test\\": \\"jest --no-cache\\",
    \\"version\\": \\"oclif-dev readme && git add README.md\\"
  },
  \\"main\\": \\"lib/index.js\\"
}
"
`;

exports[`plugin generator generates plugin with JavaScript defaults 2`] = `
"# checkup-plugin-my-plugin

Checkup plugin

[![oclif](https://img.shields.io/badge/cli-oclif-brightgreen.svg)](https://oclif.io)
[![Version](https://img.shields.io/npm/v/plugin-default.svg)](https://npmjs.org/package/plugin-default)
[![Downloads/week](https://img.shields.io/npm/dw/plugin-default.svg)](https://npmjs.org/package/plugin-default)
[![License](https://img.shields.io/npm/l/plugin-default.svg)](https://github.com/checkupjs/checkup/blob/master/package.json)

<!-- toc -->

# Usage

<!-- usage -->

# Commands

<!-- commands -->
"
`;

exports[`plugin generator generates plugin with JavaScript defaults 3`] = `
"module.exports = {
  testEnvironment: 'node',
  moduleFileExtensions: ['js', 'json'],
  coverageReporters: ['lcov', 'text-summary'],
  collectCoverageFrom: ['src/**/*.js'],
  coveragePathIgnorePatterns: ['/templates/'],
  coverageThreshold: {
    global: {
      branches: 100,
      functions: 100,
      lines: 100,
      statements: 100,
    },
  },
  testPathIgnorePatterns: ['__fixtures__'],
};
"
`;

exports[`plugin generator generates plugin with JavaScript defaults 4`] = `
"export default {};
"
`;

exports[`plugin generator generates plugin with JavaScript defaults 5`] = `
"import { Hook } from '@oclif/config';

const hook = async function({ context, tasks }) {
  
};

export default hook;
"
`;

exports[`plugin generator generates plugin with JavaScript defaults 6`] = `""`;

exports[`plugin generator generates plugin with JavaScript defaults 7`] = `
Array [
  ".gitkeep",
]
`;

exports[`plugin generator generates plugin with JavaScript defaults 8`] = `
Array [
  ".gitkeep",
]
`;

exports[`plugin generator generates plugin with JavaScript defaults 9`] = `
Array [
  ".gitkeep",
]
`;

exports[`plugin generator generates plugin with custom options 1`] = `
"{
  \\"name\\": \\"checkup-plugin-my-plugin\\",
  \\"description\\": \\"My custom plugin\\",
  \\"version\\": \\"0.0.0\\",
  \\"author\\": \\"scalvert <steve.calvert@gmail.com>\\",
  \\"dependencies\\": {
    \\"@checkup/core\\": \\"^0.0.1\\",
    \\"@oclif/command\\": \\"^1\\",
    \\"@oclif/config\\": \\"^1\\",
    \\"tslib\\": \\"^1\\"
  },
  \\"devDependencies\\": {
    \\"@checkup/test-helpers\\": \\"^0.0.1\\",
    \\"@oclif/dev-cli\\": \\"^1\\",
    \\"@types/jest\\": \\"^25.1.3\\",
    \\"@types/node\\": \\"^13\\",
    \\"jest\\": \\"^25.1.0\\",
    \\"ts-jest\\": \\"^25.2.1\\",
    \\"ts-node\\": \\"^8\\",
    \\"typescript\\": \\"^3.8\\"
  },
  \\"engines\\": {
    \\"node\\": \\">= 12.*\\"
  },
  \\"files\\": [
    \\"/lib\\",
    \\"/oclif.manifest.json\\"
  ],
  \\"keywords\\": [
    \\"checkup-plugin\\",
    \\"oclif-plugin\\"
  ],
  \\"license\\": \\"MIT\\",
  \\"oclif\\": {
    \\"devPlugins\\": [
      \\"@oclif/plugin-help\\"
    ],
    \\"hooks\\": {
      \\"register-tasks\\": \\"./lib/hooks/register-tasks\\"
    }
  },
  \\"repository\\": \\"http://github.com/scalvert/checkup-plugin-my-plugin\\",
  \\"scripts\\": {
    \\"build\\": \\"yarn clean && tsc\\",
    \\"build:watch\\": \\"yarn build -w\\",
    \\"clean\\": \\"rm -rf lib\\",
    \\"postpack\\": \\"rm -f oclif.manifest.json\\",
    \\"prepack\\": \\"rm -rf lib && tsc -b && oclif-dev manifest && oclif-dev readme\\",
    \\"test\\": \\"jest --no-cache\\",
    \\"version\\": \\"oclif-dev readme && git add README.md\\"
  },
  \\"types\\": \\"lib/index.d.ts\\",
  \\"main\\": \\"lib/index.js\\"
}
"
`;

exports[`plugin generator generates plugin with custom options 2`] = `
"# checkup-plugin-my-plugin

My custom plugin

[![oclif](https://img.shields.io/badge/cli-oclif-brightgreen.svg)](https://oclif.io)
[![Version](https://img.shields.io/npm/v/plugin-default.svg)](https://npmjs.org/package/plugin-default)
[![Downloads/week](https://img.shields.io/npm/dw/plugin-default.svg)](https://npmjs.org/package/plugin-default)
[![License](https://img.shields.io/npm/l/plugin-default.svg)](https://github.com/checkupjs/checkup/blob/master/package.json)

<!-- toc -->

# Usage

<!-- usage -->

# Commands

<!-- commands -->
"
`;

exports[`plugin generator generates plugin with custom options 3`] = `
"module.exports = {
  testEnvironment: 'node',
  moduleFileExtensions: ['ts', 'js', 'json'],
  transform: { '\\\\\\\\.ts$': 'ts-jest' },
  coverageReporters: ['lcov', 'text-summary'],
  collectCoverageFrom: ['src/**/*.ts'],
  coveragePathIgnorePatterns: ['/templates/'],
  coverageThreshold: {
    global: {
      branches: 100,
      functions: 100,
      lines: 100,
      statements: 100,
    },
  },
  testPathIgnorePatterns: ['__fixtures__'],
};
"
`;

exports[`plugin generator generates plugin with custom options 4`] = `
"{
  \\"compilerOptions\\": {
    \\"declaration\\": true,
    \\"importHelpers\\": true,
    \\"module\\": \\"commonjs\\",
    \\"strict\\": true,
    \\"target\\": \\"es2017\\",
    \\"allowSyntheticDefaultImports\\": true,
    \\"sourceMap\\": true,
    \\"outDir\\": \\"lib\\",
    \\"rootDir\\": \\"src\\",
    \\"types\\": [\\"jest\\", \\"node\\"]
  },
  \\"include\\": [\\"src/**/*\\"]
}
"
`;

exports[`plugin generator generates plugin with custom options 5`] = `
"export default {};

export * from './types';
"
`;

exports[`plugin generator generates plugin with custom options 6`] = `
"import { Hook } from '@oclif/config';

const hook: Hook<'register-tasks'> = async function({ context, tasks }: any) {
  
};

export default hook;
"
`;

exports[`plugin generator generates plugin with custom options 7`] = `""`;

exports[`plugin generator generates plugin with custom options 8`] = `
Array [
  ".gitkeep",
]
`;

exports[`plugin generator generates plugin with custom options 9`] = `
Array [
  ".gitkeep",
]
`;

exports[`plugin generator generates plugin with custom options 10`] = `
Array [
  ".gitkeep",
]
`;

exports[`plugin generator generates plugin with defaults 1`] = `
"{
  \\"name\\": \\"checkup-plugin-my-plugin\\",
  \\"description\\": \\"Checkup plugin\\",
  \\"version\\": \\"0.0.0\\",
  \\"author\\": \\"\\",
  \\"dependencies\\": {
    \\"@checkup/core\\": \\"^0.0.1\\",
    \\"@oclif/command\\": \\"^1\\",
    \\"@oclif/config\\": \\"^1\\",
    \\"tslib\\": \\"^1\\"
  },
  \\"devDependencies\\": {
    \\"@checkup/test-helpers\\": \\"^0.0.1\\",
    \\"@oclif/dev-cli\\": \\"^1\\",
    \\"@types/jest\\": \\"^25.1.3\\",
    \\"@types/node\\": \\"^13\\",
    \\"jest\\": \\"^25.1.0\\",
    \\"ts-jest\\": \\"^25.2.1\\",
    \\"ts-node\\": \\"^8\\",
    \\"typescript\\": \\"^3.8\\"
  },
  \\"engines\\": {
    \\"node\\": \\">= 12.*\\"
  },
  \\"files\\": [
    \\"/lib\\",
    \\"/oclif.manifest.json\\"
  ],
  \\"keywords\\": [
    \\"checkup-plugin\\",
    \\"oclif-plugin\\"
  ],
  \\"license\\": \\"MIT\\",
  \\"oclif\\": {
    \\"devPlugins\\": [
      \\"@oclif/plugin-help\\"
    ],
    \\"hooks\\": {
      \\"register-tasks\\": \\"./lib/hooks/register-tasks\\"
    }
  },
  \\"repository\\": \\"\\",
  \\"scripts\\": {
    \\"build\\": \\"yarn clean && tsc\\",
    \\"build:watch\\": \\"yarn build -w\\",
    \\"clean\\": \\"rm -rf lib\\",
    \\"postpack\\": \\"rm -f oclif.manifest.json\\",
    \\"prepack\\": \\"rm -rf lib && tsc -b && oclif-dev manifest && oclif-dev readme\\",
    \\"test\\": \\"jest --no-cache\\",
    \\"version\\": \\"oclif-dev readme && git add README.md\\"
  },
  \\"types\\": \\"lib/index.d.ts\\",
  \\"main\\": \\"lib/index.js\\"
}
"
`;

exports[`plugin generator generates plugin with defaults 2`] = `
"# checkup-plugin-my-plugin

Checkup plugin

[![oclif](https://img.shields.io/badge/cli-oclif-brightgreen.svg)](https://oclif.io)
[![Version](https://img.shields.io/npm/v/plugin-default.svg)](https://npmjs.org/package/plugin-default)
[![Downloads/week](https://img.shields.io/npm/dw/plugin-default.svg)](https://npmjs.org/package/plugin-default)
[![License](https://img.shields.io/npm/l/plugin-default.svg)](https://github.com/checkupjs/checkup/blob/master/package.json)

<!-- toc -->

# Usage

<!-- usage -->

# Commands

<!-- commands -->
"
`;

exports[`plugin generator generates plugin with defaults 3`] = `
"module.exports = {
  testEnvironment: 'node',
  moduleFileExtensions: ['ts', 'js', 'json'],
  transform: { '\\\\\\\\.ts$': 'ts-jest' },
  coverageReporters: ['lcov', 'text-summary'],
  collectCoverageFrom: ['src/**/*.ts'],
  coveragePathIgnorePatterns: ['/templates/'],
  coverageThreshold: {
    global: {
      branches: 100,
      functions: 100,
      lines: 100,
      statements: 100,
    },
  },
  testPathIgnorePatterns: ['__fixtures__'],
};
"
`;

exports[`plugin generator generates plugin with defaults 4`] = `
"{
  \\"compilerOptions\\": {
    \\"declaration\\": true,
    \\"importHelpers\\": true,
    \\"module\\": \\"commonjs\\",
    \\"strict\\": true,
    \\"target\\": \\"es2017\\",
    \\"allowSyntheticDefaultImports\\": true,
    \\"sourceMap\\": true,
    \\"outDir\\": \\"lib\\",
    \\"rootDir\\": \\"src\\",
    \\"types\\": [\\"jest\\", \\"node\\"]
  },
  \\"include\\": [\\"src/**/*\\"]
}
"
`;

exports[`plugin generator generates plugin with defaults 5`] = `
"export default {};

export * from './types';
"
`;

exports[`plugin generator generates plugin with defaults 6`] = `
"import { Hook } from '@oclif/config';

const hook: Hook<'register-tasks'> = async function({ context, tasks }: any) {
  
};

export default hook;
"
`;

exports[`plugin generator generates plugin with defaults 7`] = `""`;

exports[`plugin generator generates plugin with defaults 8`] = `
Array [
  ".gitkeep",
]
`;

exports[`plugin generator generates plugin with defaults 9`] = `
Array [
  ".gitkeep",
]
`;

exports[`plugin generator generates plugin with defaults 10`] = `
Array [
  ".gitkeep",
]
`;

exports[`plugin generator generates plugin with unmodified name with defaults 1`] = `
"{
  \\"name\\": \\"checkup-plugin-foo\\",
  \\"description\\": \\"Checkup plugin\\",
  \\"version\\": \\"0.0.0\\",
  \\"author\\": \\"\\",
  \\"dependencies\\": {
    \\"@checkup/core\\": \\"^0.0.1\\",
    \\"@oclif/command\\": \\"^1\\",
    \\"@oclif/config\\": \\"^1\\",
    \\"tslib\\": \\"^1\\"
  },
  \\"devDependencies\\": {
    \\"@checkup/test-helpers\\": \\"^0.0.1\\",
    \\"@oclif/dev-cli\\": \\"^1\\",
    \\"@types/jest\\": \\"^25.1.3\\",
    \\"@types/node\\": \\"^13\\",
    \\"jest\\": \\"^25.1.0\\",
    \\"ts-jest\\": \\"^25.2.1\\",
    \\"ts-node\\": \\"^8\\",
    \\"typescript\\": \\"^3.8\\"
  },
  \\"engines\\": {
    \\"node\\": \\">= 12.*\\"
  },
  \\"files\\": [
    \\"/lib\\",
    \\"/oclif.manifest.json\\"
  ],
  \\"keywords\\": [
    \\"checkup-plugin\\",
    \\"oclif-plugin\\"
  ],
  \\"license\\": \\"MIT\\",
  \\"oclif\\": {
    \\"devPlugins\\": [
      \\"@oclif/plugin-help\\"
    ],
    \\"hooks\\": {
      \\"register-tasks\\": \\"./lib/hooks/register-tasks\\"
    }
  },
  \\"repository\\": \\"\\",
  \\"scripts\\": {
    \\"build\\": \\"yarn clean && tsc\\",
    \\"build:watch\\": \\"yarn build -w\\",
    \\"clean\\": \\"rm -rf lib\\",
    \\"postpack\\": \\"rm -f oclif.manifest.json\\",
    \\"prepack\\": \\"rm -rf lib && tsc -b && oclif-dev manifest && oclif-dev readme\\",
    \\"test\\": \\"jest --no-cache\\",
    \\"version\\": \\"oclif-dev readme && git add README.md\\"
  },
  \\"types\\": \\"lib/index.d.ts\\",
  \\"main\\": \\"lib/index.js\\"
}
"
`;

exports[`plugin generator generates plugin with unmodified name with defaults 2`] = `
"# checkup-plugin-foo

Checkup plugin

[![oclif](https://img.shields.io/badge/cli-oclif-brightgreen.svg)](https://oclif.io)
[![Version](https://img.shields.io/npm/v/plugin-default.svg)](https://npmjs.org/package/plugin-default)
[![Downloads/week](https://img.shields.io/npm/dw/plugin-default.svg)](https://npmjs.org/package/plugin-default)
[![License](https://img.shields.io/npm/l/plugin-default.svg)](https://github.com/checkupjs/checkup/blob/master/package.json)

<!-- toc -->

# Usage

<!-- usage -->

# Commands

<!-- commands -->
"
`;

exports[`plugin generator generates plugin with unmodified name with defaults 3`] = `
"module.exports = {
  testEnvironment: 'node',
  moduleFileExtensions: ['ts', 'js', 'json'],
  transform: { '\\\\\\\\.ts$': 'ts-jest' },
  coverageReporters: ['lcov', 'text-summary'],
  collectCoverageFrom: ['src/**/*.ts'],
  coveragePathIgnorePatterns: ['/templates/'],
  coverageThreshold: {
    global: {
      branches: 100,
      functions: 100,
      lines: 100,
      statements: 100,
    },
  },
  testPathIgnorePatterns: ['__fixtures__'],
};
"
`;

exports[`plugin generator generates plugin with unmodified name with defaults 4`] = `
"{
  \\"compilerOptions\\": {
    \\"declaration\\": true,
    \\"importHelpers\\": true,
    \\"module\\": \\"commonjs\\",
    \\"strict\\": true,
    \\"target\\": \\"es2017\\",
    \\"allowSyntheticDefaultImports\\": true,
    \\"sourceMap\\": true,
    \\"outDir\\": \\"lib\\",
    \\"rootDir\\": \\"src\\",
    \\"types\\": [\\"jest\\", \\"node\\"]
  },
  \\"include\\": [\\"src/**/*\\"]
}
"
`;

exports[`plugin generator generates plugin with unmodified name with defaults 5`] = `
"export default {};

export * from './types';
"
`;

exports[`plugin generator generates plugin with unmodified name with defaults 6`] = `
"import { Hook } from '@oclif/config';

const hook: Hook<'register-tasks'> = async function({ context, tasks }: any) {
  
};

export default hook;
"
`;

exports[`plugin generator generates plugin with unmodified name with defaults 7`] = `""`;

exports[`plugin generator generates plugin with unmodified name with defaults 8`] = `
Array [
  ".gitkeep",
]
`;

exports[`plugin generator generates plugin with unmodified name with defaults 9`] = `
Array [
  ".gitkeep",
]
`;

exports[`plugin generator generates plugin with unmodified name with defaults 10`] = `
Array [
  ".gitkeep",
]
`;
